diff -Naurw astyle-3.4/build/cmake/InstallOptions.cmake astyle/build/cmake/InstallOptions.cmake
--- astyle-3.4/build/cmake/InstallOptions.cmake	2023-06-16 15:43:05.000000000 -0400
+++ astyle/build/cmake/InstallOptions.cmake	2023-08-21 19:03:48.881505278 -0400
@@ -1,32 +1,10 @@
-# Define install directories
-# To uninstall 'xargs rm < install_manifest.txt'
-# Default linux install prefix is /usr/local"
-# This may be modified by -DCMAKE_INSTALL_PREFIX=
-# Default Win32 install prefix is not used (C:/Program Files (x86))
-if(BUILD_SHARED_LIBS OR BUILD_STATIC_LIBS)
-    if(NOT WIN32)
-        install(TARGETS astyle DESTINATION lib)
-    endif()
-else()
-    if(WIN32)
-        set(pf86 "PROGRAMFILES(x86)")
-        set(prog_files $ENV{${pf86}})
-        if(NOT ${prog_files})
-            set(prog_files $ENV{PROGRAMFILES})
-        endif()
-        install(TARGETS astyle DESTINATION "${prog_files}/AStyle")
-        install(FILES ${DOCS} DESTINATION "${prog_files}/AStyle/doc")
-    elseif(APPLE)
-        # install to the default /usr/local/bin because of SIP restrictions
-        install(TARGETS astyle DESTINATION "${CMAKE_INSTALL_PREFIX}/bin")
-        install(FILES ${DOCS} DESTINATION "${CMAKE_INSTALL_PREFIX}/share/doc/astyle/html")
-    else()
-        # change default to /usr/bin, the same as package installs
-        if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
-            set(CMAKE_INSTALL_PREFIX "/usr")
-		endif()
-        install(TARGETS astyle DESTINATION "${CMAKE_INSTALL_PREFIX}/bin")
-        install(FILES ${DOCS} DESTINATION "${CMAKE_INSTALL_PREFIX}/share/doc/astyle/html")
-        INSTALL(FILES ${MAN} DESTINATION "${CMAKE_INSTALL_PREFIX}/man/man1")
-	endif()
-endif()
+install(TARGETS astyle
+	LIBRARY DESTINATION lib
+	ARCHIVE DESTINATION lib
+	RUNTIME DESTINATION bin
+	)
+install(FILES ${DOCS} DESTINATION share/doc/astyle/html)
+if (MAN)
+	INSTALL(FILES ${MAN} DESTINATION man/man1)
+endif (MAN)
+
diff -Naurw astyle-3.4/CMakeLists.txt astyle/CMakeLists.txt
--- astyle-3.4/CMakeLists.txt	2023-06-16 15:43:05.000000000 -0400
+++ astyle/CMakeLists.txt	2023-08-21 19:09:03.156094017 -0400
@@ -4,7 +4,7 @@
 
 include( build/cmake/Options.cmake REQUIRED )
 include( build/cmake/CompilerOptions.cmake REQUIRED )
-include( build/cmake/LinkerOptions.cmake REQUIRED )
+#include( build/cmake/LinkerOptions.cmake REQUIRED )
 include( build/cmake/InstallOptions.cmake REQUIRED )
 include( build/cmake/OutputFiles.cmake REQUIRED )
 include( build/cmake/PrintInfo.cmake REQUIRED )
