--- opennurbs/opennurbs_font.cpp	2023-09-09 13:52:46.157646417 -0400
+++ openNURBS/opennurbs_font.cpp	2023-09-09 13:43:33.716976293 -0400
@@ -8852,7 +8852,7 @@
     const wchar_t* prev_clean_family_name = nullptr;
     for (size_t i = 0; i < installed_count; i++)
     {
-#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX)
+#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_BSD)
       const ON_Font* font = installed_fonts[(int)i];
 #else
       const ON_Font* font = installed_fonts[i];
@@ -8915,7 +8915,7 @@
       InternalHashToName candidate;
       for (size_t i = 0; i < count0; i++)
       {
-#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX)
+#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_BSD)
         InternalHashToName e = a[(int)i];
 #else
         InternalHashToName e = a[i];
@@ -9642,7 +9642,7 @@
 
 void ON_Font::Dump(ON_TextLog& dump) const
 {
-  const bool bTextHash = dump.IsTextHash();;
+  const bool bTextHash = dump.IsTextHash();
 
   ON_wString s;
 
