--- opennurbs/opennurbs_polyline.cpp	2023-09-09 13:52:46.165646304 -0400
+++ openNURBS/opennurbs_polyline.cpp	2023-09-09 13:43:33.720976246 -0400
@@ -453,7 +453,7 @@
   maxN = maxN.UnitVector();
   for (size_t i = 0; i < point_count; ++i)
   {
-#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX)
+#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_BSD)
     double d = maxN * C[(unsigned int)i];
 #else
     double d = maxN * C[i];
