--- opennurbs/opennurbs_bounding_box.cpp	2023-09-09 13:52:46.153646473 -0400
+++ openNURBS/opennurbs_bounding_box.cpp	2023-09-09 13:43:33.712976340 -0400
@@ -529,7 +529,7 @@
         if ( x < -clip_plane_tolerance )
           out |= cpbit;
         cpbit <<= 1;
-        cpeqn++;;
+        cpeqn++;
       }
     }
     some_out |= out;
@@ -593,7 +593,7 @@
         if ( x < -clip_plane_tolerance )
           out |= cpbit;
         cpbit <<= 1;
-        cpeqn++;;
+        cpeqn++;
       }
     }
     some_out |= out;
@@ -657,7 +657,7 @@
         if ( x < -clip_plane_tolerance )
           out |= cpbit;
         cpbit <<= 1;
-        cpeqn++;;
+        cpeqn++;
       }
     }
     some_out |= out;
@@ -743,7 +743,7 @@
         if ( x < -clip_plane_tolerance )
           out |= cpbit;
         cpbit <<= 1;
-        cpeqn++;;
+        cpeqn++;
       }
     }
     w = xform[12]*cv[0] + xform[13]*cv[1] + xform[14]*cv[2] + xform[15];
@@ -807,7 +807,7 @@
         if ( x < -clip_plane_tolerance )
           out |= cpbit;
         cpbit <<= 1;
-        cpeqn++;;
+        cpeqn++;
       }
     }
     w = xform[12]*cv[0] + xform[13]*cv[1] + xform[14]*cv[2] + xform[15];
@@ -1047,7 +1047,7 @@
         if ( x < -clip_plane_tolerance )
           out |= cpbit;
         cpbit <<= 1;
-        cpeqn++;;
+        cpeqn++;
       }
     }
     w = xform[12]*cv[0] + xform[13]*cv[1] + xform[14]*cv[2] + xform[15]*cv[3];
@@ -1112,7 +1112,7 @@
         if ( x < -clip_plane_tolerance )
           out |= cpbit;
         cpbit <<= 1;
-        cpeqn++;;
+        cpeqn++;
       }
     }
     w = xform[12]*cv[0] + xform[13]*cv[1] + xform[14]*cv[2] + xform[15]*cv[3];
