--- opennurbs/opennurbs_3dm_properties.h	2023-09-09 13:52:46.153646473 -0400
+++ openNURBS/opennurbs_3dm_properties.h	2023-09-09 13:43:33.708976388 -0400
@@ -146,7 +146,7 @@
 {
 public:
   ON_3dmProperties() = default;
-  ~ON_3dmProperties() = default;;
+  ~ON_3dmProperties() = default;
   ON_3dmProperties(const ON_3dmProperties&) = default;
   ON_3dmProperties& operator=(const ON_3dmProperties&) = default;
 
