--- opennurbs/opennurbs_string_format.cpp	2023-09-09 13:52:46.165646304 -0400
+++ openNURBS/opennurbs_string_format.cpp	2023-09-09 13:43:33.724976199 -0400
@@ -807,7 +807,7 @@
   // CLang modifies args so a copy is required
   va_list args_copy;
   va_copy (args_copy, args);
-#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX)
+#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_BSD)
   int len = vsnprintf(buffer, buffer_capacity, format, args_copy);
 #else
   int len = vsnprintf_l(buffer, buffer_capacity, ON_Locale::Ordinal.NumericLocalePtr(), format, args_copy);
@@ -859,7 +859,7 @@
   va_list args_copy;
   va_copy (args_copy, args);
 
-#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX)
+#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_BSD)
   int len = vsnprintf(nullptr, 0, format, args_copy);
 #else
   int len = vsnprintf_l(nullptr, 0, ON_Locale::Ordinal.NumericLocalePtr(), format, args_copy);
--- opennurbs/opennurbs_string_scan.cpp	2023-09-09 13:52:46.165646304 -0400
+++ openNURBS/opennurbs_string_scan.cpp	2023-09-09 13:43:33.724976199 -0400
@@ -85,8 +85,8 @@
   va_list args
   )
 {
-#if defined(ON_COMPILER_CLANG) || defined(ON_RUNTIME_LINUX)
-#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX)
+#if defined(ON_COMPILER_CLANG) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_BSD)
+#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_BSD)
   if (nullptr == buffer || nullptr == format)
     return -1;
   return vsscanf(buffer, format, args);
@@ -159,7 +159,7 @@
   if (nullptr == args)
     return -1;
   return swscanf_l(buffer, _c_locale, format, args);
-#elif defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX)
+#elif defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_BSD)
   return swscanf(buffer, format, args);
 #else
   return swscanf(buffer, format, args);
@@ -398,8 +398,8 @@
   local_buffer[local_buffer_count++] = 0;
 
   double x = value_on_failure;
-#if defined(ON_COMPILER_CLANG) || defined(ON_RUNTIME_LINUX)
-#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX)
+#if defined(ON_COMPILER_CLANG) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_BSD)
+#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_BSD)
   if (1 == sscanf(local_buffer, "%lg", &x))
   {
     *value = x;
@@ -660,8 +660,8 @@
   local_buffer[local_buffer_count++] = 0;
 
   double x = value_on_failure;
-#if defined(ON_COMPILER_CLANG) || defined(ON_RUNTIME_LINUX)
-#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX)
+#if defined(ON_COMPILER_CLANG) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_BSD)
+#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_BSD)
   if (1 == sscanf(local_buffer, "%lg", &x))
   {
     *value = x;
