--- opennurbs/opennurbs_object_history.cpp	2023-09-09 13:52:46.161646361 -0400
+++ openNURBS/opennurbs_object_history.cpp	2023-09-09 13:43:33.720976246 -0400
@@ -31,7 +31,7 @@
   // The VALUE_TYPE enum values must never be changed
   // because the values are used to determine the parameter
   // type during file reading.  Additions can be made.
-  enum VALUE_TYPE
+  enum VALUE_TYPE: unsigned int
   {
     no_value_type         =  0,
 
