--- opennurbs/opennurbs_material.cpp	2023-09-09 13:52:46.161646361 -0400
+++ openNURBS/opennurbs_material.cpp	2023-09-09 13:43:33.720976246 -0400
@@ -5159,7 +5159,7 @@
         // map and clamp the tcs that hang over.  If the mesh
         // has edges near the texture seam, the picture will
         // still look ok.
-        float f0=0.0f, f1=0.0f, twopitc = (float)two_pi_tc;;
+        float f0=0.0f, f1=0.0f, twopitc = (float)two_pi_tc;
         //int f0cnt=0, f1cnt=0;
         if ( 1 == ftc.quad[0] ) f0 += ftc.Tx[0]; else if ( 4 == ftc.quad[0] ) f1 += twopitc-ftc.Tx[0];
         if ( 1 == ftc.quad[1] ) f0 += ftc.Tx[1]; else if ( 4 == ftc.quad[1] ) f1 += twopitc-ftc.Tx[1];
