diff -Naur flex-2.6.4/Makefile.am flex/Makefile.am
--- flex-2.6.4/Makefile.am	2017-05-03 16:16:37.000000000 -0400
+++ flex/Makefile.am	2023-09-14 10:54:03.506247328 -0400
@@ -43,10 +43,6 @@
 
 SUBDIRS = \
 	src \
-	doc \
-	examples \
-	po \
-	tests \
 	tools
 
 # Create the ChangeLog, but only if we're inside a git working directory
diff -Naur flex-2.6.4/Makefile.in flex/Makefile.in
--- flex-2.6.4/Makefile.in	2017-05-06 10:49:09.000000000 -0400
+++ flex/Makefile.in	2023-09-14 10:54:22.545929433 -0400
@@ -450,10 +450,6 @@
 
 SUBDIRS = \
 	src \
-	doc \
-	examples \
-	po \
-	tests \
 	tools
 
 all: all-recursive
